### ISSUE TITLE
Support ragged inputs to our text generation utilites

### ISSUE THREAD

**Issue by mattdangerw** [keras-team/keras-nlp#295]
For the text generation utilities listed [here](https://keras.io/api/keras_nlp/utils/), it would be nice to support ragged inputs, so we could remove the constraint that all seed text in a given batch needs to be exactly the same way.

The simplest way to do this would be to also support a ragged prompt for the generation utils.

**Comment by abheesht17** [comment]
Will be glad to take this up!

**Comment by mattdangerw** [comment]
Thanks!
